cpp/src/search/search.cu

```diff
@@ -248,83 +237,115 @@ bool contains_scalar_dispatch::operator()<cudf::dictionary32>(column_view const&
                                  : false;
 }
 
-}  // namespace
-
-namespace detail {
-bool contains(column_view const& col, scalar const& value, rmm::cuda_stream_view stream)
-{
-  if (col.is_empty()) { return false; }
-  if (not value.is_valid(stream)) { return col.has_nulls(); }
-
-  return cudf::type_dispatcher(col.type(), contains_scalar_dispatch{}, col, value, stream);
-}
-
 struct multi_contains_dispatch {
-  template <typename Element>
-  std::unique_ptr<column> operator()(column_view const& haystack,
-                                     column_view const& needles,
-                                     rmm::cuda_stream_view stream,
-                                     rmm::mr::device_memory_resource* mr)
+  template <typename Type>
+  std::enable_if_t<!is_nested<Type>(), std::unique_ptr<column>> operator()(
+    column_view const& haystack,
+    column_view const& needles,
+    rmm::cuda_stream_view stream,
+    rmm::mr::device_memory_resource* mr)
   {
-    std::unique_ptr<column> result = make_numeric_column(data_type{type_to_id<bool>()},
-                                                         haystack.size(),
-                                                         copy_bitmask(haystack),
-                                                         haystack.null_count(),
-                                                         stream,
-                                                         mr);
-
+    auto const n_rows = haystack.size();
```

Comment:
tbh, I find these kind of local variables to obscure more than they help. `haystack.size()` is unambiguous and tells me exactly what it is. When I read `n_rows` later below, I have to ask "is this the number of rows in the haystack or needles?". It's not that many more characters either.